src/platform/silabs/tracing/SilabsTracing.cpp

```diff
@@ -15,9 +15,11 @@
  *
  ******************************************************************************/
 #include "SilabsTracing.h"
+#include <cstdio> // Include the necessary header for snprintf
 #include <cstring>
 #include <lib/support/CodeUtils.h>
 #include <lib/support/PersistentData.h>
+#include <string> // Include the necessary header for std::string
```

Comment:
you have `cstring`, `string` and `cstdio`

I would have expected either `stdio.h. with string.h` or `cstdio with cstring` but not three of them no?

Reply:
cstring and cstdio are compatibility includes for C standard library to be included in C++, <string.h> is for C++ string. Let me check if there would be a clean way to ditch C++ string type here...